NadekoBot.Core/Modules/Searches/SearchesSlashCommands.cs

```diff
@@ -0,0 +1,119 @@
+using AngleSharp;
+using AngleSharp.Dom;
+using AngleSharp.Html.Dom;
+using AngleSharp.Html.Parser;
+using Discord;
+using Discord.Interactions;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using NadekoBot.Common;
+using NadekoBot.Common.Attributes;
+using NadekoBot.Common.Replacements;
+using NadekoBot.Core.Modules.Searches.Common;
+using NadekoBot.Core.Services;
+using System.Collections.Generic;
+using NadekoBot.Extensions;
+using NadekoBot.Modules.Searches.Common;
+using NadekoBot.Modules.Searches.Services;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Drawing.Processing;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using NadekoBot.Modules.Administration.Services;
+using Serilog;
+using Configuration = AngleSharp.Configuration;
+
+namespace NadekoBot.Modules.Searches
+{
+    public partial class SearchesSlash : NadekoSlashModule<SearchesService>
+    {
+        private readonly IHttpClientFactory _httpFactory;
+
+        private readonly GuildTimezoneService _tzSvc;
+        public SearchesSlash(IHttpClientFactory factory,GuildTimezoneService tzSvc)
+        {
+            _httpFactory = factory;
+            _tzSvc = tzSvc;
+        }
+
+        [NadekoSlash]
+        public async Task Weather([Summary("city","name of the city")]string city,
+        [Summary("country","optional name of country")]string country = "")
+        {
+            if (!await ValidateQuery(ctx.Channel, city).ConfigureAwait(false))
+                return;
+            string query;
+            await ctx.Interaction.DeferAsync().ConfigureAwait(false);
+            // case for no country
+            if (country == ""){
+                query = city;
+            //lookup country code using json file
```

Comment:
Since the list of countries is pretty consistent, it would be better to skip the network call and just download that data file and bundle it with the bot. It would even be fine if you hardcoded the list into this file itself (or in CountryCode.cs). In my experience hardcoding something that basically never changes is easier than trying to deal with third party flakiness.

Reply:
I added a json file